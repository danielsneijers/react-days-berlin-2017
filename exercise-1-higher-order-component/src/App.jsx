import React from 'react';
import { string } from 'prop-types';
import withUserData from './withUserData';

const App = ({ firstName = 'no name provided ðŸ˜”' }) => <h1>Hello {firstName}</h1>;

App.propTypes = {
  firstName: string.isRequired,
};

// Look, higher order components!
export default withUserData(App);
