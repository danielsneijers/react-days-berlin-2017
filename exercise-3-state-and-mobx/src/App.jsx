/* eslint-disable react/forbid-prop-types, react/require-default-props */
import React, { Component } from 'react';
import { inject, observer } from 'mobx-react';
import { object } from 'prop-types';

@inject('counterStore')
@observer
class App extends Component {
  constructor(props) {
    super(props);

    this.state = { withDecimals: false };
  }

  toggleDecimals = () => {
    this.setState({ withDecimals: !this.state.withDecimals });
  };

  render() {
    const { counterWithDecimals, counter, increaseCounter } = this.props.counterStore;
    const value = this.state.withDecimals ? counterWithDecimals : counter;

    return (
      <div>
        <h1>Counter: {value}</h1>
        <button onClick={increaseCounter}>Increase!</button>
        <button onClick={this.toggleDecimals}>Toggle decimals</button>
      </div>
    );
  }
}

App.propTypes = {
  counterStore: object,
};

export default App;
