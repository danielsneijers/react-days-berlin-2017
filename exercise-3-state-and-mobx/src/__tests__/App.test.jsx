import React from 'react';
import { mount, shallow } from 'enzyme';
import CounterStore from '../store';
import App from '../App';

describe('<App />', () => {
  let store;

  beforeEach(() => {
    store = new CounterStore();
  });

  it('renders correctly', () => {
    const tree = mount(<App counterStore={store} />);

    expect(tree).toMatchSnapshot();
  });

  it('increases the counter on button clicks', () => {
    const tree = mount(<App counterStore={store} />);
    const header = tree.find('h1');
    const increaseButton = tree.find('button').at(0);

    expect(header.text()).toBe('Counter: 0');

    increaseButton.simulate('click');
    expect(header.text()).toBe('Counter: 1');

    increaseButton.simulate('click');
    expect(header.text()).toBe('Counter: 2');
  });

  it('toggles decimals on button clicks', () => {
    const tree = mount(<App counterStore={store} />);
    const header = tree.find('h1');
    const toggleButton = tree.find('button').at(1);

    expect(header.text()).toBe('Counter: 0');

    toggleButton.simulate('click');
    expect(header.text()).toBe('Counter: 0.00');

    toggleButton.simulate('click');
    expect(header.text()).toBe('Counter: 0');
  });

  it('does not contain internal state', () => {
    const tree = shallow(<App counterStore={store} />);

    expect(tree.shallow().state()).toEqual({});
  });
});
