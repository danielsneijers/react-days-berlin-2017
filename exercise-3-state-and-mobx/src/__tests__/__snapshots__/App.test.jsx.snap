// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<inject-App-with-counterStore
  counterStore={
    CounterStore {
      "counter": 0,
    }
  }
>
  <App
    counterStore={
      CounterStore {
        "counter": 0,
      }
    }
  >
    <div>
      <h1>
        Counter: 
        0
      </h1>
      <button
        onClick={[Function]}
      >
        Increase!
      </button>
      <button
        onClick={[Function]}
      >
        Toggle decimals
      </button>
    </div>
  </App>
</inject-App-with-counterStore>
`;
